### PR TITLE
fix debug launch

### PR DESCRIPTION
**Pull request by baronfel**
### <samp>🤖 Generated by Copilot at 8b62aaa</samp>

This pull request enhances the debugging experience for F# projects in the ionide-vscode-fsharp extension. It updates the activation and debugging logic to use the `coreclr` debugger type, fixes some bugs related to the debug settings and the syntax highlighting, and improves the logging and type compatibility of the `Debugger` module. It also updates the `RELEASE_NOTES.md` file to document these changes.

### <samp>🤖 Generated by Copilot at 8b62aaa</samp>

> _We are the F# warriors, we debug with coreclr_
> _We activate and launch our code, we resolve our variables_
> _We log and box and unbox, we format with node.js_
> _We fix the bugs and highlight the syntax, we write the release notes_

🐛🚀📝

### WHY
Instead of declaring that Ionide knew how to handle 'coreclr' launch types, all we really needed to do was have our debug adapter launch when 'coreclr' launch types are being queried. This fixes the breaking behavior for users.

### <samp>🤖 Generated by Copilot at 8b62aaa</samp>

*  Add activation events for coreclr debugger in `package.json` ([link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-7902ed52e6f39559068a38e3c223d7f7d7d0b0977aee33d659db648092a62b15L9-R11))
*  Remove unused debuggers section from `package.json` ([link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-7902ed52e6f39559068a38e3c223d7f7d7d0b0977aee33d659db648092a62b15L829-L834))
*  Update logging statements in `Debugger.fs` to use node.js console formatting syntax ([link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-5cd78ecc280ba189382f47bebe9e3eff8a6c6e86439fdf867f4be407b9ef728fL340-R340), [link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-5cd78ecc280ba189382f47bebe9e3eff8a6c6e86439fdf867f4be407b9ef728fL380-R378), [link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-5cd78ecc280ba189382f47bebe9e3eff8a6c6e86439fdf867f4be407b9ef728fL389-R387))
*  Replace Seq.collect with List.collect in `Debugger.fs` for performance and readability ([link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-5cd78ecc280ba189382f47bebe9e3eff8a6c6e86439fdf867f4be407b9ef728fL346-R347))
*  Apply hack to convert promise to thenable in `Debugger.fs` to workaround type mismatch issue ([link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-5cd78ecc280ba189382f47bebe9e3eff8a6c6e86439fdf867f4be407b9ef728fL363-R370))
*  Add release notes for version 7.5.4 in `RELEASE_NOTES.md` ([link](https://github.com/ionide/ionide-vscode-fsharp/pull/1871/files?diff=unified&w=0#diff-68d24fa81558aae3d8c59e2aa57a4fa719ea3b04d7fa14beff45f16f00858f50R1-R5))

telplin-debug

### CODE REVIEW

File: release/package.json [hunk 1]
```diff
@@ -6,7 +6,9 @@
   },
   "activationEvents": [
     "onCommand:workbench.action.tasks.runTask",
+    "onDebugInitialConfigurations:coreclr",
     "onDebugDynamicConfigurations:coreclr",
+    "onDebugResolve:coreclr",
```
Comment (baronfel) [inline release/package.json:11]: These are nice-to-haves - making sure that we are initialized whenever coreclr launch types are used

File: release/package.json [hunk 2]
```diff
@@ -826,12 +828,6 @@
       "title": "F#",
       "type": "object"
     },
-    "debuggers": [
-      {
-        "label": "Ionide LaunchSettings",
-        "type": "coreclr"
-      }
-    ],
```
Comment (baronfel) [inline release/package.json:834]: This is the core fix - no longer overwriting

File: src/Components/Debugger.fs [hunk 3]
```diff
@@ -337,14 +337,14 @@ module Debugger =
             override x.provideDebugConfigurations(folder: option<WorkspaceFolder>, token: option<CancellationToken>) =
                 let generate () =
                     promise {
-                        logger.Info $"Evaluating launch settings configurations for workspace '%A{folder}'"
+                        logger.Info("Evaluating launch settings configurations for %O", folder)
```
Comment (baronfel) [inline src/Components/Debugger.fs:340]: Fixing bugged logging

File: src/Components/Debugger.fs [hunk 4]
```diff
@@ -360,24 +360,22 @@ module Debugger =
...
+                |> unbox
 
             override x.resolveDebugConfiguration
                 (
                     folder: option<WorkspaceFolder>,
                     debugConfiguration: DebugConfiguration,
                     token: option<CancellationToken>
                 ) =
-                logger.Info $"Evaluating launch settings configurations for workspace2 '{folder}'"
+                logger.Info("Resolving launch settings configuration %O for %O", debugConfiguration, folder)
```
Comment (baronfel) [inline src/Components/Debugger.fs:378]: One interesting thing here - VSCode calls this method with a null debugConfiguration if you press `f5` and no launch configuration has been configured for this workspace before - maybe we should find the current open editor (if any) or selected project in the tree (if any?) and force that task?

**Comment by baronfel** [comment]
This fixes #1870

**Review by TheAngryByrd — APPROVED** [review]

**Pull request merged**